content/blog/2016-05-09-when-to-use-varbit-chunks.md

```diff
@@ -0,0 +1,159 @@
+---
+title: When (not) to use varbit chunks
+created_at: 2016-05-09
+kind: article
+author_name: Björn “Beorn” Rabenstein
+---
+
+The embedded time-series database (TSDB) of the Prometheus server organizes the
+raw sample data of each time series in chunks of constant 1024 bytes size. In
+addition to the raw sample data, a chunk contains some meta-data, which allows
+the selection of a different encoding for each chunk. The most fundamental
+distinction is the encoding version. You select the version for newly
+created chunks via the command line flag
+`-storage.local.chunk-encoding-version`. So far, there were only two supported
+versions: 0 for the original delta encoding, and 1 for the improved
+double-delta encoding. With release
+[0.18.0](https://github.com/prometheus/prometheus/releases/tag/0.18.0), we
+added version 2, which is another variety of double-delta encoding. We call it
+_varbit encoding_ because it involves a variable bit-width per sample within
+the chunk. While version 1 is superior to version 0 in almost every aspect,
+there is a real trade-off between version 1 and 2. This blog post will help you
+to make that decision. Version 1 remains the default encoding, so if you want
+to try out version 2 after reading this article, you have to select it
+explicitly via the command line flag. There is no harm in switching back and
+forth, but note that chunks will not change their encoding version once they
```

Comment:
I don't understand.

GC will never rewrite chunks. Only transcoding might happen occasionally, but even that won't change the encoding version.